packages/application-config/src/load-config.ts

```diff
@@ -37,52 +37,97 @@ const loadJsModule: LoaderSync = (filePath) => {
-    `${moduleName}.mjs`,
-    `${moduleName}.ts`,
-  ],
-  loaders: {
-    noExt: defaultLoaders['.json'],
-    '.js': loadJsModule,
-    '.cjs': loadJsModule,
-    '.mjs': loadJsModule,
-    '.ts': loadJsModule,
-  },
-});
+const getExplorer = (configFileName: string) => {
+  return cosmiconfigSync(configFileName, {
+    // Restrict the supported file formats / names
+    searchPlaces: [
+      `.${configFileName}rc`,
+      `.${configFileName}.json`,
+      `.${configFileName}.js`,
+      `.${configFileName}.cjs`,
+      `.${configFileName}.mjs`,
+      `.${configFileName}.ts`,
+      `${configFileName}.json`,
+      `${configFileName}.js`,
+      `${configFileName}.cjs`,
+      `${configFileName}.mjs`,
+      `${configFileName}.ts`,
+    ],
+    loaders: {
+      noExt: defaultLoaders['.json'],
+      '.js': loadJsModule,
+      '.cjs': loadJsModule,
+      '.mjs': loadJsModule,
+      '.ts': loadJsModule,
+    },
+  });
+};
+
+const customApplicationExplorer = getExplorer('custom-application-config');
+const customViewExplorer = getExplorer('custom-view-config');
+
+// const moduleName = 'custom-application-config';
+// const explorer = cosmiconfigSync(moduleName, {
+//   // Restrict the supported file formats / names
+//   searchPlaces: [
+//     `.${moduleName}rc`,
+//     `.${moduleName}.json`,
+//     `.${moduleName}.js`,
+//     `.${moduleName}.cjs`,
+//     `.${moduleName}.mjs`,
+//     `.${moduleName}.ts`,
+//     `${moduleName}.json`,
+//     `${moduleName}.js`,
+//     `${moduleName}.cjs`,
+//     `${moduleName}.mjs`,
+//     `${moduleName}.ts`,
+
+//     'custom-view.config.json',
+//     'custom-view.config.js',
+//     'custom-view.config.cjs',
+//     'custom-view.config.mjs',
+//     'custom-view.config.ts',
+//   ],
+//   loaders: {
+//     noExt: defaultLoaders['.json'],
+//     '.js': loadJsModule,
+//     '.cjs': loadJsModule,
+//     '.mjs': loadJsModule,
+//     '.ts': loadJsModule,
+//   },
+// });
 
 export const getConfigPath = () => {
-  const configFile = explorer.search();
-  if (!configFile) {
-    throw new Error(
-      `Missing or invalid Custom Application configuration file.`
-    );
+  const customApplicationConfigFile = customApplicationExplorer.search();
+  const customViewConfigFile = customViewExplorer.search();
+
+  if (!customApplicationConfigFile && !customViewConfigFile) {
+    throw new Error(`Missing or invalid configuration file.`);
   }
-  return configFile.filepath;
+  return (
+    customApplicationConfigFile?.filepath || customViewConfigFile?.filepath
+  );
 };
 
 const loadConfig = (
   applicationPath: string
-): JSONSchemaForCustomApplicationConfigurationFiles => {
-  const configFile = explorer.search(applicationPath);
+): {
+  filepath: string;
+  config: JSONSchemaForCustomApplicationConfigurationFiles;
+} => {
+  const customApplicationConfigFile =
+    customApplicationExplorer.search(applicationPath);
+  const customViewConfigFile = customViewExplorer.search(applicationPath);
```

Comment:
We try to load either config file for Custom Apps and Custom Views as only one of them will be found.
You won't have both in your project.

Reply:
I didn't expect a scenario where a user have both configuration files in the same application.
Maybe you can share your concern about how that could be possible?

In any case, given such a use case, why we would be preferring one over the other? We would not know what the users' application is supposed to implement.